op-node/cmd/genesis/cmd.go

```diff
@@ -180,39 +160,32 @@ var Subcommands = cli.Commands{
 				return errors.New("missing l2-allocs")
 			}
 
-			if l1RPC != "" {
-				client, err := ethclient.Dial(l1RPC)
-				if err != nil {
-					return fmt.Errorf("cannot dial %s: %w", l1RPC, err)
-				}
+			// Retrieve SystemConfig.startBlock()
+			client, err := ethclient.Dial(l1RPC)
+			if err != nil {
+				return fmt.Errorf("cannot dial %s: %w", l1RPC, err)
+			}
 
-				if config.L1StartingBlockTag == nil {
-					l1StartBlock, err = client.BlockByNumber(context.Background(), nil)
-					if err != nil {
-						return fmt.Errorf("cannot fetch latest block: %w", err)
-					}
-					tag := rpc.BlockNumberOrHashWithHash(l1StartBlock.Hash(), true)
-					config.L1StartingBlockTag = (*genesis.MarshalableRPCBlockNumberOrHash)(&tag)
-				} else if config.L1StartingBlockTag.BlockHash != nil {
-					l1StartBlock, err = client.BlockByHash(context.Background(), *config.L1StartingBlockTag.BlockHash)
-					if err != nil {
-						return fmt.Errorf("cannot fetch block by hash: %w", err)
-					}
-				} else if config.L1StartingBlockTag.BlockNumber != nil {
-					l1StartBlock, err = client.BlockByNumber(context.Background(), big.NewInt(config.L1StartingBlockTag.BlockNumber.Int64()))
-					if err != nil {
-						return fmt.Errorf("cannot fetch block by number: %w", err)
-					}
-				}
+			systemConfigAbi := snapshots.LoadSystemConfigABI()
+			contract := bind.NewBoundContract(config.SystemConfigProxy, *systemConfigAbi, client, client, client)
+			callOpts := &bind.CallOpts{Context: context.Background()}
+			var result []interface{}
+			err = contract.Call(callOpts, &result, "startBlock")
```

Comment:
I'd love to avoid hardcoding the "startBlock" function name here if possible

Reply:
Updated this code to use a more lightweight approach. Still hardcoding the function name, but it seems a worthy tradeoff in this case to save lots of lines of unused code